pynitrokey/cli/update.py

```diff
@@ -45,7 +40,42 @@
 def update(serial, yes, local_firmware_server, alpha):
     """Update Nitrokey key to latest firmware version."""
 
-    update_url = 'https://update.nitrokey.com/'
-    print('Please use {} to run the firmware update'.format(update_url))
-    return
+    #update_url = 'https://update.nitrokey.com/'
+    #print('Please use {} to run the firmware update'.format(update_url))
+    #return
+
+
+    latest_release_url = "https://github.com/Nitrokey/nitrokey-fido2-firmware/releases/latest"
+
+    web_data = requests.get(latest_release_url)
+
+    res = re.findall(r'a href="([^"]+)"', web_data.text)
+    _url = None
+    for item in res:
+        if "firmware" in item and "download" in item and item.endswith(".json"):
+            _url = item
+            break
```

Comment:
There is an API on Github, which allows to request the links by json directly, no need to parse the web page.
Please take a look at Nitrokey Start firmware update handling, where it was used. While I do not see problems with it right now, it might not be future-proof (e.g. the page content would be generated entirely dynamically), hence better to use official API, which already aggregates the needed data.

Reply:
jup jup, no discussion, lazy me :smile: